nixos/tests/nixos-test-driver/multi-os/qemu-vm.nix

```diff
@@ -0,0 +1,352 @@
+              - `mapped-xattr`: some of the file attributes like uid, gid, mode bits and link target are stored as file attributes
+              - `mapped-file`: the attributes are stored in the hidden .virtfs_metadata directory. Directories exported by this security model cannot interact with other unix tools
+              - `none`: same as "passthrough" except the sever won't report failures if it fails to set file attributes like ownership
+            '';
+          };
+        }
+      );
+      default = { };
+      example = {
+        my-share = {
+          source = "/path/to/be/shared";
+          target = "/mnt/shared";
+        };
+      };
+      description = ''
+        An attributes set of directories that will be shared with the
+        virtual machine using VirtFS (9P filesystem over VirtIO).
+        The attribute name will be used as the 9P mount tag.
+      '';
+    };
+    # FIXME (copied from qemu-vm.nix)
+    virtualisation.interfaces = mkOption {
+      default = { };
+      example = {
+        enp1s0.vlan = 1;
+      };
+      description = ''
+        Network interfaces to add to the VM.
+      '';
+      type =
+        with types;
+        attrsOf (submodule {
+          options = {
+            vlan = mkOption {
+              type = types.ints.unsigned;
+              description = ''
+                VLAN to which the network interface is connected.
+              '';
+            };
+
+            assignIP = mkOption {
+              type = types.bool;
+              default = false;
+              description = ''
+                Automatically assign an IP address to the network interface using the same scheme as
+                virtualisation.vlans.
+              '';
+            };
+          };
+        });
+    };
+    system.name = mkOption {
+      type = types.str;
+      # FIXME: not accurate?
+      default = name;
+    };
+  };
+  config = {
+    # FIXME
+    system.build.vm =
+      hostPkgs.runCommand "nixos-vm"
+        {
+          preferLocalBuild = true;
+          meta.mainProgram = "run-${config.system.name}-vm";
+        }
+        ''
+          mkdir -p $out/bin
+          ln -s ${config.system.build.toplevel} $out/system
+          ln -s ${hostPkgs.writeScript "run-nixos-vm" startVM} $out/bin/run-${config.system.name}-vm
+        '';
+    virtualisation.qemu.options = [
+      "-enable-kvm" # ??
+      "-machine"
+      "q35,accel=kvm"
+      "-usb" "-device" "usb-kbd" "-device" "usb-tablet" "-device" "usb-tablet"
+      "-cpu" "Penryn,kvm=on,vendor=GenuineIntel,+invtsc,vmware-cpuid-freq=on,+ssse3,+sse4.2,+popcnt,+avx,+aes,+xsave,+xsaveopt,check"
+      "-device" "usb-ehci,id=ehci"
+      "-device" "nec-usb-xhci,id=xhci"
+      "-global" "nec-usb-xhci.msi=off"
+      "-device" ''isa-applesmc,osk="ourhardworkbythesewordsguardedpleasedontsteal(c)AppleComputerInc"''
```

Comment:
This string has, of course, been the subject of substantial legal dispute. I don’t know if there’s any consensus position on whether there is a problematic copyright or DMCA aspect to including it here. Psystar *did* lose their case, though.

(It’s only relevant for `x86_64-darwin` QEMU hacks, not `aarch64-darwin` using the Virtualization framework, so I think it probably isn’t relevant these days, but I figured we would require end users to supply this string as an assertion that they accept that it’s their responsibility to comply with the EULA.)